rpc/v2/server.go

```diff
@@ -110,7 +111,10 @@ func (s *Server) RegisterName(name string, rcvr interface{}) error {
 // 2. supports notifications (pub/sub)
 // 3. supports request batches
 func (s *Server) ServeCodec(codec ServerCodec) {
+	ctx, cancel := context.WithCancel(context.Background())
+
 	defer func() {
+		cancel()
```

Comment:
Nitpicking, but you're inverting the object creation/defer order here a bit.

Usually the cleanest way to do defers is:
 - Make object A
 - Defer cleanup of A
 - Make object B
 - Defer cleanup of B

You here kind of did
 - Make codec
 - Make context
 - Defer cleanup of both, but you need to take care of the order manually.

It would be a bit cleaner to lave the old defer as is at the top of the method, and *after* the old defer do

```
ctx, cancel := context.WithCancel(context.Background())
defer cancel()
```